libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1140,13 +1188,10 @@ void WaveClip::SetSequenceStartTime(double startTime)
 
 double WaveClip::GetSequenceEndTime() const
 {
-    const auto numSamples = GetNumSamples();
-
-    double maxLen = GetSequenceStartTime() + (numSamples + GetAppendBufferLen()).as_double() / mRate;
-    // JS: calculated value is not the length;
-    // it is a maximum value and can be negative; no clipping to 0
-
-    return maxLen;
+   const auto numSamples = GetNumSamples();
+   double maxLen = GetSequenceStartTime() +
+                   numSamples.as_double() * GetStretchRatio() / mRate;
```

Comment:
The correction by `GetAppendBufferLen` was removed.  Was that your omission or an intended bug fix?

Reply:
I only remember I had a reason at the time, I'm sorry I should have written this down. Now it seems wrong to me the way it is, and is inconsistent with `GetPlayEndSample()`.
In any case, I really would like to clarify the meaning of the append buffer length.
I guess this append buffer is here for optimization, to reduce calls to SQL. If that's right, shouldn't these not-yet-committed samples _always_ be reported? `GetNumSamples()` doesn't account for it, and is used in several places where I don't see why it shouldn't (e.g., in `TimeToSequenceSamples`, non-committed samples are not accounted for).